### DISCUSSION TITLE
Periodic Scheduled Downloads for Offline Access to Documents from Paperless-ngx Server

### DISCUSSION THREAD

**Discussion by fire1ce** [LeoKlaus/Paperparrot#150] [Feature Requests]
I would like to propose the implementation of a feature that enables users to schedule periodic downloads of documents from the Paperless-ngx server directly to their iPhone. This functionality would be particularly beneficial for scenarios where there is no ability to access the remote Paperless-ngx server, ensuring that users can still retrieve and interact with their documents.

Use Case:
The primary use case for this feature is to facilitate uninterrupted access to documents even when the server is in an "offline" state or temporarily inaccessible. By pre-downloading the necessary documents to the iPhone, users can ensure they have the required information at their fingertips, regardless of the server's availability. This adds an additional layer of reliability and user convenience, especially for those who frequently find themselves in areas with unstable or no internet connection.

This feature, if implemented, would undoubtedly enhance the user experience even further.

**Comment by LeoKlaus** [comment]
Hi again,

Paperparrot already allows you to keep an offline copy of your documents. You can enable downloads in the Settings within Paperparrot under "App Settings"->"Download Documents".
Tapping "Download all documents" will start the download process.
![Simulator Screenshot - iPhone 15 Pro - 2023-11-01 at 10 59 09](https://github.com/LeoKlaus/Paperparrot/assets/66492595/a49fa3ca-85de-417d-98ae-75c42bfad7ac)

While this is enabled, Paperparrot will try to download all new documents on each subsequent launch, so your documents stay up-to-date.

While offline, all "passive" activities work (i.e. search, custom views, looking at and sharing/exporting documents). Actions that require a server connection like uploading or editing documents will only work when connected to a server.

**Comment by fire1ce** [comment]
Thanks for the fast reply, i misunderstood this function but o be “one time” my bad. closing the suggestion.

**Reply by LeoKlaus** [reply]
No problem, I might have to improve the wording.
